blueprints/tags.md

```diff
@@ -0,0 +1,421 @@
+<!--
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+# Tags
+
+## Problem Description
+Tags are a fairly long-requested feature; to be able to associate a key word or
+phrase with a Delivery Service, or a user. Today, some organizations are abusing
+the overloaded "description" properties of a Delivery Service to provide
+comma-separated key words and phrases, essentially inventing their own tags by
```

Comment:
I imagine they may be using the "description" field for key-value pairs as well. If the tag is alphanumeric how should someone represent a key-value pair?

Reply:
They won't, which just means tags won't replace those kinds of description abuse.

Can you give me an example of something that might be used as a key-value pair in a description field?

Reply:
I've relaxed the naming constraints to allow `_` and `=`, but I wouldn't recommend actually using tags for key/value pairs. If there's a key that's needed on Delivery Services, I'd rather see it as a proper property of DSes. Otherwise implementing that would essentially mean creating a model of Delivery Service that is mutable at runtime, which seems like a bad idea.

Reply:
Key-value tags would allow operators to essentially create whatever informational fields that they may want on a delivery service that might not make sense for everyone else's installation. For instance, maybe Company A wants to create "service categories" for delivery services in order to group them and do certain things with those groups outside the scope of TO. Company B might not care about or even want service categories, so instead of it just always being an empty, unused field in Company B's delivery services, it could exist purely in Company A's world via key-value tags.

Reply:
But if you want to put Delivery Services into groups of service categories, say either the "Foo" service category or the "Bar" category, you could use key-value pairs to say `serviceCategory: "Foo" | "Bar"`, or you could just give them either the "Foo" or the "Bar" tag. The only difference is enforced exclusivity, and some semantic meaning (which tags are not supposed to have).

If there's a need to store some data in a way that's validated - either structurally or semantically - and it's not useful outside of your organization, such that you don't want to contribute it to the project, then perhaps a plugin for an endpoint like `deliveryservices_categories` could be useful?